lib/puppet/parser/scope.rb

```diff
@@ -816,12 +845,12 @@ def effective_symtable(use_ephemeral)
   #
   # @param varname [String] The variable name to which the value is assigned. Must not contain `::`
   # @param value [String] The value to assign to the given variable name.
-  # @param options [Hash] Additional options, not part of api.
+  # @param options [Hash] Additional options, not part of api and no longer used.
   #
   # @api public
   #
-  def []=(varname, value, options = {})
-    setvar(varname, value, options = {})
```

Comment:
Despite what the comment says, this is a breaking change, right? Previously, you could do `scope['foo', 'bar', :append => true)` and change how the variables were defined. Now the `options` argument is ignored.